test/src/tests-vault.ts

```diff
@@ -0,0 +1,271 @@
+import { Batch } from '@mparticle/event-models';
+import { expect } from 'chai';
+import Vault from '../../src/vault';
+
+describe.only('Vault', () => {
+    afterEach(() => {
+        window.localStorage.clear();
+    });
+
+    describe('#storeBatches', () => {
+        it('should store an array of batches as a hashmap', () => {
+            const batch1 = {
+                foo: 'bar',
+                source_request_id: 'item-123',
+            };
+            const batch2 = {
+                foo: 'bar',
+                source_request_id: 'item-456',
+            };
+
+            const batches = ([batch1, batch2] as unknown) as Batch[];
+
+            const vault = new Vault('test-key-store-batches');
+
+            const expectedContents = { 'item-123': batch1, 'item-456': batch2 };
```

Comment:
Minor, but I suggest moving `expectedContents` to immediately before the expect assertion on line 29 to not interrupt the actions the tests are taking.